### PR TITLE
5320/johanna remove loinc code from device type

### LINKED ISSUE #5320: Remove `DeviceType.LoincCode` from frontend
### Background
We have recently added the new field name `DeviceTestPerformedLoincCode` to store supported disease and loinc code, which makes the standalone fields obsolete and deprecated.

### Action requested
Remove any usage of `DeviceType.LoincCode` from the front end.

### Acceptance Criteria

- LoincCode on DeviceType should not longer be used in any front end query or mutation.

### PR DESCRIPTION
**Pull request by johanna-skylight**
# FRONTEND PULL REQUEST

## Related Issue

- Resolves #5320 

## Changes Proposed
_Changes in Device Type for the admin interface_
- Device Form now uses the types from the graphql generated file and we no longer need to use the local defined `interface Device`.
- In the form data type `testOrderedLoincCode` was marked as optional to match the graphql endpoint definition but the field is still required during validation so nothing has changed for the user.
- `inferLoincCode` method was removed.
- All instances where the `loincCode` was set as a field for the device were removed across multiple components.
- `getDeviceFromDeviceType` method was trimmed down to just map more directly to the device information retrieve from the server.
- Unit testing was updated for multiple components.
- Device related graphql operations were updated to make the field `SupportedDiseaseTestPerformedInput` mandatory and to remove `loincCode`.

_Test Results_
- `GetFacilityResultsForCsvWithCount` query operation was updated to reflect the latest structure of the deviceType object .

_Device Lookup details_
- The Test performed code area was updated to list multiple testOrdered codes and not only covid.
- graphql operation `getDeviceTypesForLookup` was updated to receive the testOrdered information through the field `supportedDiseaseTestPerformed`.

_GraphQL changes in the backend_

- LoincCode was removed from the Input Types that are being used in the mutation resolvers.
- DeviceType model is set to accept null for loincCode (The removal of the field in the DB model will be done in following ticket).
- DeviceTypeService sets the loincCode field as null until next ticket removes the field in the DB.
- Types in main.graphql `CreateDeviceType` `UpdateDeviceType` `DeviceType` were updated to not return `loincCode` and make `supportedDiseaseTestPerformed` mandatory.
- Unit testing was updated.

## Additional Information

- During the implementation of this ticket I hit a weird point where local frontend unit testing were running fine in local but failing in the pipeline or vice versa. As part of the changes you will see small fixes to unit testing of components not really related to the implementation of this ticket.

## Testing
_Deployed in dev7_

- Add a device
- Edit a device
- Add the device to a facility
- Perform a test with the device
- Check the result looks good
- Check the device with the lookup tool

## Screenshots / Demos
Only UI change noticeable to the user is in the device lookup tool

Screenshot 2023-03-20 at 12 55 14 PM

### CODE REVIEW

**Review by emyl3 — COMMENTED** [review]
If we make the both the backend and frontend changes at the same time, I remember emma sharing we would run into an issue if a user has the new backend code and the old frontend for X number of minutes. Is that still true? 🤔

**Comment by johanna-skylight** [comment]
> If we make the both the backend and frontend changes at the same time, I remember emma sharing we would run into an issue if a user has the new backend code and the old frontend for X number of minutes. Is that still true? 🤔

This is true but not for all the graphql changes. In the case where the UI is trying to create or update a device (mutations) the  old frontend will continue to send the loincCode in the request but once the new backend code gets to prod these fields will just be ignored so backwards compatibility on that front is all good.

For the cases where the UI is getting data to display (query) the old frontend would indeed expect the fields (loincCode, swabType) to render correctly if the query is including that value as part of the response object. I have added back the loincCode and the swabType in the DeviceType definition but I have marked these fields as deprecated. 

Let me know if this sounds good to you! @emyl3

File: backend/src/test/resources/graphql-test/device-type-query.graphql [hunk 1]
```diff
@@ -4,7 +4,5 @@ query listDevices {
         internalId
         manufacturer
         model
-        loincCode
-        swabType
```
[the remark this answers is not in the document]
Comment (emyl3) [reply backend/src/test/resources/graphql-test/device-type-query.graphql:8]: OO got it! Thank you

File: frontend/src/app/testResults/operations.graphql [hunk 2]
```diff
@@ -54,7 +54,10 @@ query GetFacilityResultsForCsvWithCount(
         name
         manufacturer
         model
-        swabType
```
Comment (emyl3) [inline frontend/src/app/testResults/operations.graphql:57]: thank you for the cleanup 🧹

**Review by emyl3 — APPROVED** [review]
LGTM! Added and edited devices on dev7 and tried the device code lookup. 👍

**Review by BobanL — APPROVED** [review]
Changes look good! Thanks for the clean up :D

**Pull request merged**